### PR TITLE
MSSM ggH/ggA powheg cards

### PR DESCRIPTION
**Pull request by Tommaso93**
Dear GEN conveners,
these are the gluon fusion powheg cards in the MSSM HiggsToMuMu analysis with UL pdfs.

Best,
Tommaso

### CODE REVIEW

File: bin/Powheg/production/2017/13TeV/Higgs/MSSM/ggA_MuMu_NNPDF31_13TeV/gg_A_MSSM_h_5f_13TeV.input [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+!!!!!!!!!!! MSSM OS parameter  !!!!!!!!!
+FHslhaflag 0  ! relevant when scheme=2 . If 0 read FeynHiggs input param from powheg-fh.in, otherwise from powheg-fh.slha
+FHdecaywidth 1 ! 1 = use the decay width provided by FeynHiggs
+! bottommass 4.75d0 ! If not set, the bottom OS mass is calculated with a 1-loop scheme change formula.
+
+
+!!!!!!!!!!! MSSM DRbar parameter  !!!!!!!!!
+mssmmbatmh 0 ! 1 = mb(mh) is used (from mssm-param-mh.slha), 0 = mb(M_susy) is used
+pdfreweight 1       ! PDF reweighting
+storeinfo_rwgt 1    ! store weight information
```
Comment (agrohsje) [inline bin/Powheg/production/2017/13TeV/Higgs/MSSM/ggA_MuMu_NNPDF31_13TeV/gg_A_MSSM_h_5f_13TeV.input:102]: Please remove. Will be automatically handled.

File: bin/Powheg/production/2017/13TeV/Higgs/MSSM/ggH_MuMu_NNPDF31_13TeV/powheg.input_template [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+!!!!!!!!!!! MSSM OS parameter  !!!!!!!!!
+FHslhaflag 0  ! relevant when scheme=2 . If 0 read FeynHiggs input param from powheg-fh.in, otherwise from powheg-fh.slha
+FHdecaywidth 1 ! 1 = use the decay width provided by FeynHiggs
+! bottommass 4.75d0 ! If not set, the bottom OS mass is calculated with a 1-loop scheme change formula.
+
+
+!!!!!!!!!!! MSSM DRbar parameter  !!!!!!!!!
+mssmmbatmh 0 ! 1 = mb(mh) is used (from mssm-param-mh.slha), 0 = mb(M_susy) is used
+pdfreweight 1       ! PDF reweighting
+storeinfo_rwgt 1    ! store weight information
```
Comment (agrohsje) [inline bin/Powheg/production/2017/13TeV/Higgs/MSSM/ggH_MuMu_NNPDF31_13TeV/powheg.input_template:102]: Please remove.

File: bin/Powheg/production/2017/13TeV/Higgs/MSSM/ggA_MuMu_NNPDF31_13TeV/gg_A_MSSM_h_5f_13TeV.input (hunk 1 above)
Comment (Tommaso93) [reply bin/Powheg/production/2017/13TeV/Higgs/MSSM/ggA_MuMu_NNPDF31_13TeV/gg_A_MSSM_h_5f_13TeV.input]: Done

Comment (Tommaso93) [reply bin/Powheg/production/2017/13TeV/Higgs/MSSM/ggA_MuMu_NNPDF31_13TeV/gg_A_MSSM_h_5f_13TeV.input]: reminder

File: bin/Powheg/production/2017/13TeV/Higgs/MSSM/ggH_MuMu_NNPDF31_13TeV/powheg.input_template (hunk 2 above)
Comment (Tommaso93) [reply bin/Powheg/production/2017/13TeV/Higgs/MSSM/ggH_MuMu_NNPDF31_13TeV/powheg.input_template]: reminder

**Comment by agrohsje** [comment]
Let's converge in the bbH thread. @covarell for completeness.

**Comment by Tommaso93** [comment]
Dear @agrohsje, thank you for the help! The problem of the condor jobs stucked was solved. I also cleaned the cards, as requested.

**Comment by agrohsje** [comment]
Can you say more about the solution?  Did changing the seed help?

**Comment by Tommaso93** [comment]
Yes, changing the seed was the solution! After changing that in the run_pwg_condor script, the stucked condor jobs started to work!

**Comment by Tommaso93** [comment]
Do you need any additional info?

**Comment by Tommaso93** [comment]
If everything is ok, can you please close the PR?

**Comment by agrohsje** [comment]
I see e.g. ew 0 or neg weights disabled. May I ask, was the choice of settings studied? Where are they coming from? Do you have a reference talk with plots ?

**Comment by Tommaso93** [comment]
Hi @agrohsje, the cards are taken from the ggHToTauTau analysis. They have then been studied and adapted with the parameters relevant to the MSSM e.g. mA and tanB.

- Concerning the **ew** parameter set to 0, after a discussion with the analysis group and an expert, it has to be set to 1 only for h/H (the pseudoscalar A production does not have those diagrams because of the absence of the relevant couplings).
I will change that in the next commit.

- For the negative weights, I contacted one of the HExtended conveners and we looked at some of the samples produced by TauTau group (example here: https://cms-pdmv.cern.ch/mcm/requests?prepid=HIG-RunIISummer15wmLHEGS-04930&page=0&shown=127). They have enabled the negweights due to an high number of them, and so I will do it as well in the next commit. 

The rest of the card seems ok to us.

**Comment by Tommaso93** [comment]
Hi @agrohsje, the cards are taken from the ggHToTauTau analysis. They have then been studied and adapted with the parameters relevant to the MSSM e.g. mA and tanB.

- Concerning the ew parameter set to 0, after a discussion with the analysis group and an expert, it has to be set to 1 only for h/H (the pseudoscalar A production does not have those diagrams because of the absence of the relevant couplings). I will change that in the next commit.

- For the negative weights, I contacted one of the HExtended conveners and we looked at some of the samples produced by TauTau group (example here: https://cms-pdmv.cern.ch/mcm/requests?prepid=HIG-RunIISummer15wmLHEGS-04930&page=0&shown=127). They also seem to have turned off the negative weights. To be sure, I tried to produce a sample including the negative weights: the negative weight fraction is:   2.4085218468077055E-002. It is not negligible, therefore I believe it is best to include them. What do you think?

The rest of the card seems ok to us.

**Comment by Tommaso93** [comment]
Hi @agrohsje, I decided to test the negative weights by producing 1000 events and looking at the number of events with negative weights. On 12 events there are negative weights. I also compared the distributions of muon_pt for both gridpacks (with and without the negative weights) and they appear to differ a bit.

**Comment by Tommaso93** [comment]
If you agree, I can commit the changes

**Comment by Tommaso93** [comment]
Dear @agrohsje, if you have any additional question, otherwise let me know if I can commit the changes mentioned above. I need to start creating the requests as soon as possible. Also for the other PR #2916

**Comment by agrohsje** [comment]
I am really sorry @Tommaso93. I saw your mails and feedback. Thanks so much. Give me a bit to review and check.

**Comment by Tommaso93** [comment]
Thank you @agrohsje, let me know!

**Comment by agrohsje** [comment]
@Tommaso93  agree with what you write about the neg. weights. please adjust. thanks for following up on the ew parameters.

File: bin/Powheg/production/2017/13TeV/Higgs/MSSM/ggA_MuMu_NNPDF31_13TeV/powheg.input_template [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+!	               11  ZZ
+!	               12  gamma gamma
+hwidth 10d0	
+hdecaywidth 0	! 1 read Higgs decay width from HDECAY file
+zerowidth 0 ! Control if the Higgs boson is to be produced on-shell or not: 1 = On-Shell; 0 = Off-shell with Breit-Wigner
+bwshape 1	! BW shape: 1 running width BW, 2 fixed width BW
+masswindow_high 5d0 ! number of widths around hmass in the BW for an off-shell Higgs boson  upper limit
+masswindow_low 5d0  ! number of widths around hmass in the BW for an off-shell Higgs boson  lower limit
+ew 0                     ! ew = 0 disable EW corrections - ew = 1 enable EW corrections
+fastew 1                   ! fast ew corrections evaluations by mass sampling
```
Comment (agrohsje) [inline bin/Powheg/production/2017/13TeV/Higgs/MSSM/ggA_MuMu_NNPDF31_13TeV/powheg.input_template:21]: why is this 1 while ew is 0 ? maybe not relevant in this case?

File: bin/Powheg/production/2017/13TeV/Higgs/MSSM/ggH_MuMu_NNPDF31_13TeV/powheg.input_template [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+!		       10  W+W-
+!	               11  ZZ
+!	               12  gamma gamma
+hwidth 10d0	
+hdecaywidth 0	! 1 read Higgs decay width from HDECAY file
+zerowidth 0 ! Control if the Higgs boson is to be produced on-shell or not: 1 = On-Shell; 0 = Off-shell with Breit-Wigner
+bwshape 1	! BW shape: 1 running width BW, 2 fixed width BW
+masswindow_high 5d0 ! number of widths around hmass in the BW for an off-shell Higgs boson  upper limit
+masswindow_low 5d0  ! number of widths around hmass in the BW for an off-shell Higgs boson  lower limit
+ew 0                     ! ew = 0 disable EW corrections - ew = 1 enable EW corrections
```
Comment (agrohsje) [inline bin/Powheg/production/2017/13TeV/Higgs/MSSM/ggH_MuMu_NNPDF31_13TeV/powheg.input_template:20]: didn't you want to change this to 1 ?

File: bin/Powheg/production/2017/13TeV/Higgs/MSSM/ggh_MuMu_NNPDF31_13TeV/powheg.input_template [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+!		       10  W+W-
+!	               11  ZZ
+!	               12  gamma gamma
+hwidth 10d0	
+hdecaywidth 0	! 1 read Higgs decay width from HDECAY file
+zerowidth 0 ! Control if the Higgs boson is to be produced on-shell or not: 1 = On-Shell; 0 = Off-shell with Breit-Wigner
+bwshape 1	! BW shape: 1 running width BW, 2 fixed width BW
+masswindow_high 5d0 ! number of widths around hmass in the BW for an off-shell Higgs boson  upper limit
+masswindow_low 5d0  ! number of widths around hmass in the BW for an off-shell Higgs boson  lower limit
+ew 0                     ! ew = 0 disable EW corrections - ew = 1 enable EW corrections
```
Comment (agrohsje) [inline bin/Powheg/production/2017/13TeV/Higgs/MSSM/ggh_MuMu_NNPDF31_13TeV/powheg.input_template:20]: same here

File: bin/Powheg/production/2017/13TeV/Higgs/MSSM/ggA_MuMu_NNPDF31_13TeV/powheg.input_template (hunk 3 above)
Comment (Tommaso93) [reply bin/Powheg/production/2017/13TeV/Higgs/MSSM/ggA_MuMu_NNPDF31_13TeV/powheg.input_template:21]: Yes, it is not relevant in this case

**Comment by Tommaso93** [comment]
Ok @agrohsje, I adjusted the cards.

File: bin/Powheg/production/2017/13TeV/Higgs/MSSM/ggH_MuMu_NNPDF31_13TeV/powheg.input_template (hunk 4 above)
Comment (Tommaso93) [reply bin/Powheg/production/2017/13TeV/Higgs/MSSM/ggH_MuMu_NNPDF31_13TeV/powheg.input_template]: Done

File: bin/Powheg/production/2017/13TeV/Higgs/MSSM/ggh_MuMu_NNPDF31_13TeV/powheg.input_template (hunk 5 above)
Comment (Tommaso93) [reply bin/Powheg/production/2017/13TeV/Higgs/MSSM/ggh_MuMu_NNPDF31_13TeV/powheg.input_template]: Done

**Pull request merged**